community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CompilerLibrary.scala

```diff
@@ -45,6 +47,15 @@ class CompilerLibrary(factory: CompilerFactory) {
     } else compiler
   }
 
+  def clearCaches(): Long = {
```

Comment:
What value is being returned here? The max of cleared entries feels weird to me. But maybe a comment could clarify that. Also, it seems all the caches that share this method (and probably others) could inherit the same trait.